coalib/processes/Processing.py

```diff
@@ -421,6 +429,31 @@ def instantiate_processes(section,
             bear_runner_args)
 
 
+def uncache_if_unloaded_bears(initial_count,
+                              loaded_count,
+                              files_dict,
+                              cache,
+                              log_printer):
+    """
+    Checks if any bear could not be instantiated because of a missing
+    dependency and uncaches current section's cached files
+    to permit a future execution of the failing bear.
+
+    :param initial_count: The number of bears that the section should execute.
+    :param loaded_count:  The number of bears that were actually
+                          loaded with success.
+    :param files_dict:    The files this section runs on. Theese are the files
+                          that will be removed from the cache.
```

Comment:
These files are removed from the cache only in case there was a problem instantiating a bear right.